packages/platform-android/src/commands/runAndroid/index.ts

```diff
@@ -135,34 +135,74 @@ function buildAndRun(args: Flags) {
     packageName,
   );
   const adbPath = getAdbPath();
-  if (args.deviceId) {
+  let devices;
+
+  if (!args.deviceId) {
+    devices = adb.getDevices(adbPath);
+  }
+
+  if (
+    devices &&
+    devices.length > 1 &&
+    devices.filter(Boolean).length > 1 &&
```

Comment:
I would personally prefer a bit more verbose check here. I don't understand what's the purpose of `filter(Boolean)`, so maybe a simple inline comment would do?

Also, I think that `devices.length > 1` is not needed if we have the `devices.filter(Boolean).length > 1` already in place.